### DISCUSSION TITLE
How to pass command line arguments to bash/python script task template

### DISCUSSION THREAD

**Discussion by yp-zhu** [semaphoreui/semaphore#2262] [Q&A] [answered]
I tried to setup bash/python script template, but I could not get the arguments to pass to the actual script. I added "arg" in the "Args" section, also tried manually entering the CLI args json array ["arg1","arg2"]. but script never pickup those. I also watch the actual command line it launches; I do not see any arguments in the command line. it just like 'bash myscript.sh'. Did I miss anything? Anyone has example with template setup include arguments to the script (bash/python)?
I searched Q&A section and did not see any answers related to this topic.
Here is the test bash script:
```
#!/bin/bash 
echo "Hello, we are testing bash here"
echo "command line args [$@]"
echo "done"
```

Thanks,

**Comment by lafayetteduarte** [comment]
+1 
Same question for powershell. 
https://github.com/semaphoreui/semaphore/discussions/2279

**Reply by lafayetteduarte** [reply]
In my case ( powershell s far), the workaround was :
Use env variables as a means of passing arguments inside the script. 
far from ideal but works

**Reply by yp-zhu** [reply]
In my case the environment variables worked. but using arguments is much easier and be able to pass during the runtime.

**Comment by yp-zhu** [comment]
Found the issue in the code. services/tasks/LocalJob.go function "getShellArgs" Line#175. It only adds the script name (t.Template.Playbook) to the args, but it never append the t.Template.Arguments and t.Task.Arguments to the "args" array.

**Reply by lafayetteduarte** [reply]
This pr seems to fix this. 
https://github.com/semaphoreui/semaphore/pull/2288

**Comment by SinanAEM** [comment]
Python templates are not working for me, I'm using Sempahore 2.10.22-e44910d-1721658299

I wrote a simple python script to print out arguments that the script receives :

import sys

num_args = len(sys.argv) - 1  

print(f'Number of arguments passed: {num_args}')

# Print the arguments
if num_args > 0:
    print('Arguments:')
    for i, arg in enumerate(sys.argv[1:], start=1):
        print(f'Argument {i}: {arg}')
else:
    print('No arguments were passed.')


and then I tried to run the template multiple times using 'args' , 'CLI args in task' and I see there are no arguments being fetched from Semaphore 

![2](https://github.com/user-attachments/assets/e429a7cb-f98f-401f-86f2-d0f69ad0e29b)
![1](https://github.com/user-attachments/assets/64cfd5fb-601a-4c99-922f-4b4ed704a7f2)
![3](https://github.com/user-attachments/assets/c081a15e-194d-400c-bc90-1dfd085aa068)

**Reply by SinanAEM** [reply]
although I installed Semaphore from RPM I did notice that there is no ".go" app for python in db_lib folder in the Repo:
https://github.com/semaphoreui/semaphore/tree/develop/db_lib

**Comment by ottacom** [comment]
I my case since I installed semaphore from the binary file  and I'm lazy :-) ... the easy way to turn around the problems it has been:

#!/bin/bash
export $1
export $2
export $3
export $4
#HOSTNANE="$1"
#DOMAIN="$2"
#NETWORK="$3"
#PROFILE="$4"
echo $VM_HOSTNAME
echo $DOMAIN
echo $NETWORK
echo $PROFILE

**Comment by fiftin** [comment, marked as answer]
Fixed here: https://github.com/semaphoreui/semaphore/pull/2689
